src/stencils1d.jl

```diff
@@ -0,0 +1,48 @@
+
+abstract type Order1D end
+abstract type Extrapolate1D end
+
+struct Linear <: Order1D end
+
+struct Flat <: Extrapolate1D end
+struct LinearExtrapolation <: Extrapolate1D end
```

Comment:
Could you add some docstrings explaining the algorithms and expected behavior for these cases?

Reply:
Docstrings are added.